php/libraries/Database.class.inc

```diff
@@ -307,10 +307,23 @@ class Database
     {
         $retVal = array();
         foreach ($set as $key => $val) {
-            if ($val === 'Unrated') {
-                $retVal[$key] = '';
+            // Only call htmlspecialchars on strings, or else an error occurs.
+            if (is_string($val)) {
+                if ($val === 'Unrated') {
```

Comment:
since this is going to major maybe it's time to get rid of this "Unrated" hack and see what breaks (I'm not sure what if was ever there for, but if anything I suspect the imaging browser..)